### ISSUE TITLE
Missing support for issue form templates

### ISSUE THREAD

**Issue by Fs00** [slapperwan/gh4a#1159]
Recently, GitHub introduced a new type of issue templates called "[issue forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms)" that are being increasingly used by repo maintainers.
Those templates are [YAML files](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms) (instead of traditional Markdown files) that are used to generate a form that the user fills when creating a new issue.
Currently, the app ignores this kind of templates.

Implementing them in the app seems a pretty challenging task. We would probably need to parse the YAML file in order to generate the UI for the form and then, upon send, convert the filled form into a Markdown issue body that can be submitted via API.
Obviously it would be easier if GitHub exposed some dedicated APIs for issue forms, I haven't investigated yet.

**Comment by KaKi87** [comment]
Hello,

Since this doesn't seem to be happening anytime soon, would it be possible for OctoDroid, at least, to redirect users to the browser when trying to create an issue on a repo than uses forms, preventing them to create empty ones, and then getting rejected, or worse, judged for using a third-party client ?

Thanks

**Comment by KaKi87** [comment]
@maniac103 :pray:

**Comment by Cwpute** [comment]
This is thoroughly needed.
Contributing to projects you're not accustomed with through Octodroid curently make it seem like they don't use templates, so when you are required to use ones on these repositories, you can get turned down for not using them. Not mentioning the maintainers themselves having to deal with contributors not using templates, while having the best intentions.

**Comment by Cwpute** [comment]
At least a warning saying this features isn't yet implementel would be useful !
i just fell for it and an issue i opened through Octodroid got closed due to not following template guidelines.

**Comment by ccuser44** [comment]
Bump!

**Comment by KaKi87** [comment]
That would be functionally nice yes, but the *very* least, as I suggested, is to redirect to the browser.

**Comment by ccuser44** [comment]
> That would be functionally nice yes, but the very least, as I suggested, is to redirect to the browser.

Yes. Also the functionality that I provided is extremely simple to implement too.

**Comment by maniac103** [comment]
> Also the functionality that I provided is extremely simple to implement too.

Sorry to be blunt, but: if that's the case, we're extremely happy about a pull request!

**Comment by Fs00** [comment]
Yeah, a contribution from someone else would be much appreciated. I don't plan to work on this.
@KaKi87's suggestion looks the most reasonable way forward IMO, I think that an issue form transformed into Markdown would offer a worse UX than compiling the same form via browser. Not to mention that there are issue form elements like dropdowns that don't map well to Markdown.

**Comment by KaKi87** [comment]
> a contribution from someone else would be much appreciated

I would if only I tolerated typed languages 😅

> I don't plan to work on this

May I ask then, how do you handle this issue on your side ? Do you always open repo issues in browser instead of your app ? Or always only on desktop ?

**Comment by Fs00** [comment]
I rarely open issues using the app, most of the time I do it on a PC. I understand though that for other people - who use the app differently than I do - this issue can be frustrating.
However, given the limited time I have to contribute to the app, I'd rather not spend it in fixing bugs that don't affect me that much and aren't straightforward to address (like this one).

**Comment by uli-on** [comment]
Perhaps something more simple, put up a warning about "issue template in use, go use a browser for opening the issue"? All subsequent posts are OK.
